cmake/modules/CompileWarnings.cmake

```diff
@@ -1,9 +1,16 @@
 macro(wpilib_target_warnings target)
+    set(WARNING_FLAGS
+        -Wall
+        -pedantic
+        -Wextra
+        -Wno-unused-parameter
+        ${WPILIB_TARGET_WARNINGS}
+    )
+    if(NOT NO_WERROR)
+        set(WARNING_FLAGS "${WARNING_FLAGS}" -Werror)
+    endif()
```

Comment:
This should be moved inside the if statement since it's only valid for non-MSVC compilers.